salt/daemons/masterapi.py

```diff
@@ -314,14 +315,16 @@ def check_signing_file(self, keyid, signing_file):
             log.warning('Wrong permissions for %s, ignoring content', signing_file)
             return False
 
-        with salt.utils.files.fopen(signing_file, 'r') as fp_:
-            for line in fp_:
-                line = line.strip()
-                if line.startswith('#'):
-                    continue
-                else:
-                    if salt.utils.stringutils.expr_match(keyid, line):
-                        return True
+        mtime = os.path.getmtime(signing_file)
+        if self.signing_file.get(signing_file, {}).get('mtime') != mtime:
+            self.signing_files.setdefault(signing_file, {})['mtime'] = mtime
+            with salt.utils.files.fopen(signing_file, 'r') as fp_:
+                self.signing_files[signing_file]['data'] = [
+                    line for line in [line.strip() for line in fp_] if not line.strip().startswith('#')
+                ]
+        if any(salt.utils.stringutils.expr_match(keyid, line) for line
+               in salt.signing_files[signing_file].get('data', [])):
+            return True
```

Comment:
```suggestion
            return bool(any(salt.utils.stringutils.expr_match(keyid, line) for line
                        in salt.signing_files[signing_file].get('data', [])))
```

Reply:
Yes, good idea, except `any()` already returns a bool, so wrapping it in `bool()` is redundant.

Reply:
Right...